gql/transport/requests.py

```diff
@@ -4,21 +4,31 @@
 from graphql.execution import ExecutionResult
 from graphql.language.printer import print_ast
 
-from .http import HTTPTransport
 
-
-class RequestsHTTPTransport(HTTPTransport):
-    def __init__(self, url, auth=None, use_json=False, timeout=None, **kwargs):
+class RequestsHTTPTransport(object):
```

Comment:
Since you bypassed HTTPTransport and deleted that file I think it makes sense to rename this class HTTPTransport now for simplicity.

But maybe you can add an abstract base class "Transport" with the execute method for both LocalSchemaTranport and HTTPTransport. Later it can be also used for type checking the transport argument of the Client.

Also, docstrings for these classes would be nice to have. You can mention that HTTPTransport uses requests and the POST method for instance.

Reply:
That's a good idea! The only thing I don't understand at this point is the existance of **LocalSchemaTransport** and **RequestsHttpTransport** as I have no idea what's the main difference between them.

Also, by looking at some features from #24, I do see extra transport classes were suggested (and added) like as **BatchHttpTransport**, **SessionHttpTransport** and so on.

A quick search on Google leads me to think those are _extra transport layers for GraphQL_ but not sure about this.

Cheers!

Reply:
The LocalschemaTransport uses a local, executable GraphQL schema instead of querying a remote GraphQL server.

Reply:
Extra transport classes should then all inherit from the abstract base Transport class.

Reply:
In a future version (when we require Python 3.8+), maybe we can use a [protocol](https://www.python.org/dev/peps/pep-0544/) instead of an abstract base class.

Reply:
I have made the changes, at the end, created the `Transport` file which stores the abstract class. Also added tons of docstrings and tested locally with sphinx for hyperlinks and such stuff.

Reply:
Big thanks and kudos for working on this!